src/App.tsx

```diff
@@ -15,71 +15,87 @@ export const goods = [
+export class App extends React.Component {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  setGood = (event: React.MouseEvent<HTMLButtonElement>) => {
+    const goodName = event.currentTarget.closest('tr')
+      ?.querySelector('.is-vcentered')
+      ?.textContent;
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+    this.setState({ selectedGood: goodName });
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    if (this.state.selectedGood === goodName) {
+      this.setState({ selectedGood: '' });
+    }
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  removeGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    const goodIsSelectedMessage = (
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood}
+        {' is selected'}
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+        <button
+          data-cy="ClearButton"
+          type="button"
+          className="delete ml-3"
+          onClick={this.removeGood}
+        />
+      </h1>
+    );
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+    return (
+      <main className="section container">
+        <h1 className="title">
+          {
+            selectedGood
+              ? goodIsSelectedMessage
+              : 'No goods selected'
+          }
+        </h1>
+
+        <table className="table">
+          <tbody>
+
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={selectedGood === good
+                  ? 'has-background-success-light'
+                  : ''}
```

Comment:
use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally